workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityList.cs

```diff
@@ -0,0 +1,268 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Profiling;
+using UnityEditor;
+using UnityEditor.UIElements;
+using UnityEngine.UIElements;
+
+namespace Improbable.Gdk.Debug.WorkerInspector
+{
+    public class EntityList : VisualElement
+    {
+        public delegate void EntitySelected(EntityId entityId);
+
+        public EntitySelected OnEntitySelected;
+
+        private World world;
+        private EntitySystem entitySystem;
+
+        private readonly ListView listView;
+
+        private readonly Entities entities = new Entities();
+
+        private int lastViewVersion;
+
+        public EntityList()
+        {
+            const string uxmlPath = "Packages/io.improbable.gdk.debug/WorkerInspector/Templates/EntityList.uxml";
+            var template = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(uxmlPath);
+            template.CloneTree(this);
+
+            listView = this.Q<ListView>();
+            listView.itemHeight = 24;
+            listView.makeItem = () => new Label();
+            listView.bindItem = BindItem;
+            listView.onSelectionChanged += OnSelectionChanged;
+            listView.itemsSource = entities.Data;
+
+            var searchField = this.Q<ToolbarSearchField>();
+            searchField.RegisterCallback<ChangeEvent<string>>(OnSearchFieldChanged);
+        }
+
+        public void Update()
```

Comment:
What happens if the selected world is destroyed? `world.IsCreated == false`?

Reply:
This always happens after the world selector updates, which in turn will fire off a `OnWorldChanged(null)` event. So by the time we get here, the state is all resolved